### PR TITLE
Fix for issue 7641: Wrong path to TeXstudio

### PR DESCRIPTION
**Pull request by Gan-Cheng**
Fixes [#7641](https://github.com/JabRef/jabref/issues/7641)

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

<h3>Bug:</h3>

No matter where the extended applications are, the default path of them (taken texstudio as example) would be set as `C:\Program Files (x86)\TeXstudio\texstudio.exe`. 

<h3>The way to fix: </h3>

I verify the existence of extended applications under related default path. If there is no `texstudio.exe` under `C:\Program Files (x86)\TeXstudio\` and `C:\Program Files\TeXstudio\`, the function *detectProgramPath()* will return String value "", namely the default path will remain unset.

### CODE REVIEW

File: src/main/java/org/jabref/gui/desktop/os/Windows.java [hunk 1]
```diff
@@ -27,13 +27,36 @@ public void openFile(String filePath, String fileType) throws IOException {
...
+
+    private String getProgramPath(String programName, String directoryName, String progFiles) {
+        String programPath;
         if ((directoryName != null) && !directoryName.isEmpty()) {
-            return Path.of(progFiles, directoryName, programName + DEFAULT_EXECUTABLE_EXTENSION).toString();
+            programPath = Path.of(progFiles, directoryName, programName + DEFAULT_EXECUTABLE_EXTENSION).toString();
+        } else {
+            programPath = Path.of(progFiles, programName + DEFAULT_EXECUTABLE_EXTENSION).toString();
+        }
+        File program = new File(programPath);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/desktop/os/Windows.java:55]: You can use Files.exist(path)

File: src/main/java/org/jabref/gui/desktop/os/Windows.java [hunk 2]
```diff
@@ -27,13 +27,36 @@ public void openFile(String filePath, String fileType) throws IOException {
...
+        String programPath;
+        if (progFiles != null) {
+            programPath = getProgramPath(programName, directoryName, progFiles);
+            if (programPath != null) {
+                return programPath;
+            }
         }
+
+        progFiles = System.getenv("ProgramFiles");
+        System.out.println(progFiles);
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/gui/desktop/os/Windows.java:39]: Probably a left over from debugging. If you need debug output use LOGGER.debug(...)

**Review by Siedlerchr — CHANGES_REQUESTED** [review]

**Comment by Siedlerchr** [comment]
Just some minor code changes, but otherwise lgtm!

File: src/main/java/org/jabref/gui/desktop/os/Windows.java (hunk 2 above)
Comment (Gan-Cheng) [reply src/main/java/org/jabref/gui/desktop/os/Windows.java:39]: emmm，I was too careless. Thanks for your suggestion!

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**